### PR TITLE
Add release notes

### PR DESCRIPTION
**Pull request by sleto-it**
This PR adds the release notes in the doc. This will help users better tracking releases and list of changes.
The content is taken from the `valkey` repo, `00-RELEASENOTES` files (for the branches `8.0` and `7.2`).

Links to the fixing PRs have been included (when the PR number was mentioned in the `00-RELEASENOTES`). 
Links to some GitHub Issues have been included as well (when the Issue number was mentioned in the `00-RELEASENOTES`).

A ToC was included in the page to help navigation.

Release notes have been linked in the main `Docs` menu. Related PR in `valkey-io.github.io`: https://github.com/valkey-io/valkey-io.github.io/pull/140.

**Preview below**

1  adding_release_notes-toc

---

2  adding_release_notes-preview

---

3  adding_release_notes-link-to-prs

### CODE REVIEW

File: topics/release-notes.md [hunk 1]
```diff
@@ -0,0 +1,1653 @@
...
+
+* Correctly expose client infomation in the slowlog when running commands in scripts. ([#805])
+* Fix a bug where lastbgsave_status was unecessarily being updated when dual
+  channel replication failed. ([#811])
+* Fix a bug which caused dual channel replication to get stuck because of a TLS
+  issue. ([#837])
+* Fix a bug which caused bouncing `-REDIRECT` messages during a FAILOVER ([#871])
+* Avoid re-establishing replication to a node that is already the primary in `CLUSTER REPLICATE`. ([#884])
+* Fix `CLUSTER SETSLOT` with `BLOCK` to only wait for replicas that are online. ([#879])
+* Fix `valkey-cli` to make source node ignores NOREPLICAS when doing the last CLUSTER SETSLOT ([#928])
```
Comment (nastena1606) [inline topics/release-notes.md:148]:
```suggestion
* Fix `valkey-cli` to make source node ignore NOREPLICAS when doing the last CLUSTER SETSLOT ([#928])
```

File: topics/release-notes.md [hunk 2]
```diff
@@ -0,0 +1,1653 @@
...
+#### <a id="800-rc1-reliability-improvements-core"></a>Reliability Improvements - Core
+
+* Limit tracking custom errors (e.g. from Lua) while allowing normal errors
+  to be tracked ([#500], Redis#13141)
+* Manage maximum number of new connections per cycle to prevent connection
+  storms. (Redis#12178)
+
+#### <a id="800-rc1-reliability-improvements-cluster"></a>Reliability Improvements - Cluster
+
+* Reduce fail-over time in Valkey cluster when multiple sequential fail-overs
```
Comment (nastena1606) [inline topics/release-notes.md:309]:
```suggestion
* Reduce failover time in Valkey cluster when multiple sequential fail-overs
```

Comment (nastena1606) [reply topics/release-notes.md:309]: The sentence seems incomplete. Can we amend it a bit? when multiple sequential failovers -> during multiple sequential failovers?

File: topics/release-notes.md [hunk 3]
```diff
@@ -0,0 +1,1653 @@
...
+* Optimize linear search of WAIT and WAITAOF when unblocking the client. ([#787])
+* Move prepareClientToWrite out of loop for lrange command to remove redundant calls. ([#860])
+* Optimize various commands by using sdsAllocSize instead of sdsZmallocSize. ([#923])
+* Using intrinsics to optimize counting HyperLogLog trailing bits. ([#846])
+* Free client's MULTI state when it becomes dirty to reduce memory usage. ([#961])
+* Optimize the fast path of SET if the expiration time has already expired. ([#865])
+
+#### <a id="800-rc2-compatibility-improvements"></a>Compatibility Improvements
+
+* Add 4 sentinel command `GET-PRIMARY-ADDR-BY-NAME`, `PRIMARY`, `PRIMARIES`, and
```
Comment (nastena1606) [inline topics/release-notes.md:166]:
```suggestion
* Add 4 Sentinel command `GET-PRIMARY-ADDR-BY-NAME`, `PRIMARY`, `PRIMARIES`, and
```

File: topics/release-notes.md [hunk 4]
```diff
@@ -0,0 +1,1653 @@
...
+#### <a id="800-rc1-compatibility-improvements-core"></a>Compatibility Improvements - Core
+
+* Derive RDB and module child process names based on server start name for
+  compatibility. ([#454])
+* Update server identity in `serverPanic` output based on `extended-redis-compatibility`
+  config. ([#415])
+
+#### <a id="800-rc1-compatibility-improvements-sentinel"></a>Compatibility Improvements - Sentinel
+
+* Accept `redis-sentinel` to start Valkey in sentinel mode. ([#731])
```
Comment (nastena1606) [inline topics/release-notes.md:248]:
```suggestion
* Accept `redis-sentinel` to start Valkey in Sentinel mode. ([#731])
```

File: topics/release-notes.md [hunk 5]
```diff
@@ -0,0 +1,1653 @@
...
+* Fix `valkey-check-aof` misidentifying data in manifest format as MP-AOF.
+  (Redis#12958)
+* Fix `valkey-cli` to respect the `--count` option without requiring
+  `--pattern`. (Redis#13092)
+* Fix `valkey-benchmark` to distribute operations across all slots owned by
+  a node in cluster mode. (Redis#12986)
+
+#### <a id="800-rc1-internal-codebase-improvements"></a>Internal Codebase Improvements
+
+* Enable debug asserts for cluster and sentinel tests. ([#588])
```
Comment (nastena1606) [inline topics/release-notes.md:417]:
```suggestion
* Enable debug asserts for cluster and Sentinel tests. ([#588])
```

File: topics/release-notes.md [hunk 6]
```diff
@@ -0,0 +1,1653 @@
...
+  ([#693], Fixes [Issue #619])
+* Ensure only primary nodes with slots can mark another node as failed ([#634])
+* Improve MEET command reliability under link failures to maintain cluster
+  membership symmetry ([#461])
+* Allow single primary node to mark potentially failed replica as FAIL in
+  single-shard cluster (Redis#12824)
+
+#### <a id="726-ga-bug-fixes-sentinel"></a>Bug fixes - Sentinel
+
+* Accept redis-sentinel to start Valkey in sentinel mode ([#731], Fixes [Issue #719])
```
Comment (nastena1606) [inline topics/release-notes.md:498]:
```suggestion
* Accept redis-sentinel to start Valkey in Sentinel mode ([#731], Fixes [Issue #719])
```

File: topics/release-notes.md [hunk 7]
```diff
@@ -0,0 +1,1653 @@
...
+
+* Fix module event name to maintain Redis compatibility. ([#289])
+* Fix issue where Redis symlinks were created with the wrong name. ([#282])
+
+### <a id="724-rc1-valkey-released-tue-09-apr-2024"></a>Valkey 7.2.4 RC1 - Released Tue 09 Apr 2024
+
+Upgrade urgency **LOW**.
+
+Initial release of Valkey with API compatibility for OSS
+Redis 7.2.4. This release is based off of OSS Redis 7.2.4, with additional
```
Comment (nastena1606) [inline topics/release-notes.md:545]:
```suggestion
Redis 7.2.4. This release is based on OSS Redis 7.2.4, with additional
```

**Comment by zuiderkwast** [comment]
This means we'll have to commit release notes to the doc repo every time we do a new release? It adds some extra work to the release process.

I would prefer that the website logic automatically would take the release notes from each release.

@stockholmux WDYT?

**Comment by stockholmux** [comment]
@zuiderkwast Yeah, I'd rather avoid committing the release notes to the docs repo for a variety of reasons.

Release notes are currently linked on every non-rc release from the [releases page for each version](https://valkey.io/download/releases/), but admittedly that isn't quite obvious and misses anything put in RC releases.

GitHub Actions could be used to merge all this stuff together using minimal metadata on the website side (effectively adding a new element to array with each release). The website is quite good at _including_ files inside the template, so we could do something like `/release-notes/8.0.0`, `/release-notes/8.0.0-rc2`, `/release-notes/8.0.0-rc1` , etc. with a single version's release notes.

This is subtly different from what's being proposed, which is one giant page. (The giant page is also kind of an antipattern anyway).

**Comment by zuiderkwast** [comment]
@stockholmux I like this suggestion.

**Comment by stockholmux** [comment]
@sleto-it would you mind creating an issue on https://github.com/valkey-io/valkey-io.github.io so we can track getting this built? I'm glad to roll it into your PR https://github.com/valkey-io/valkey-io.github.io/pull/140 as well.

**Comment by sleto-it** [comment]
Hi @stockholmux,
Sure, and thanks. Done: https://github.com/valkey-io/valkey-io.github.io/issues/145

**Pull request closed without merging**